assessments-service/src/main/java/com/transformuk/hee/tis/assessment/service/service/impl/SpecificationFactory.java

```diff
@@ -109,4 +118,49 @@ public static Specification isBetween(String attribute, int min, int max) {
   public static Specification isBetween(String attribute, double min, double max) {
     return (root, query, cb) -> cb.between(root.get(attribute), min, max);
   }
+
+  /**
+   * Returns a list of values from the provided column filter.
+   * Correctly handles assessment LocalDate fields, including dot-notated sub entity fields.
+   *
+   * @param cf The column filter to use
+   * @return A list of values
+   */
+  public static List<Object> getDateAwareValuesFromColumnFilter(ColumnFilter cf) {
+    List<Object> valuesList;
+    try {
+      String[] subEntity = StringUtils.split(cf.getName(), DOT);
+      Field field;
+      if (subEntity.length == 1) {
+        field = Assessment.class.getDeclaredField(subEntity[0]);
+      } else {
+        switch (subEntity[0].toLowerCase()) {
+          case "revalidation":
+            field = Revalidation.class.getDeclaredField(subEntity[1]);
+            break;
+          case "outcome":
+            field = AssessmentOutcome.class.getDeclaredField(subEntity[1]);
+            break;
+          case "detail":
+            field = AssessmentDetail.class.getDeclaredField(subEntity[1]);
+            break;
+          default:
+            throw new NoSuchFieldException(subEntity[0]);
+        }
+      }
+      if (field.getType().equals(LocalDate.class)) {
+        //dates need to be handled differently to strings / numbers
+        valuesList = cf.getValues().stream()
+            .map(c -> LocalDate.parse(c.toString()))
```

Comment:
Just notice the The `LocalDate.parse(CharSequence text)` obtains an instance of LocalDate from a text string such as 2007-12-03, which is also implied in the test. However, the date format we use across TIS is like 03/12/2007, so I'm a little wondering should we normalise the data format? 
I guess there might be some conversion of the date format in Generic Upload to make it YYYY-MM-DD, so I probably didn't take this into account.

Reply:
That is an interesting point, I wonder if there is a localisation issues that might be hiding here? MySQL works natively with dates in YYYY-MM-DD format, so I simply assumed (uh-oh!) that would be fine. 
I see in generic-upload the XLS objects just have column mappings that expect a Date field; testing by uploading a file where a date column has been reformatted to a number (which gives values like '44260.00') triggers the error:
`File uploaded failed validation : Date missing or incorrect format on mandatory field (reviewDate) Date missing or incorrect format on mandatory field (reviewDate). File was not uploaded.`
The code that produces this is here: https://github.com/Health-Education-England/TIS-GENERIC-UPLOAD/blob/e7121a1a8210bce14e5263bbbb946e04f7c591e8/generic-upload-service/src/main/java/com/transformuk/hee/tis/genericupload/service/api/validation/FileValidator.java#L160 . 
I guess by the time it reaches the context of this ticket the Date has been resolved, but it would be interesting to know how portable the code is for other timezones!

Reply:
Yeah, the Date format in the DB are all yyyy-MM-DD, while on the UI and bulk upload, the business related date fields are all in format dd/MM/yyyy. 
So I guess when someone is going to use this client or add the `ReviewDate` columnfilter from the UI, he/she would expect to send this parameter in format dd/MM/yyyy. (maybe some docs could be added somewhere for explanation?)

Reply:
I don't think the REST API is really used by front-end users directly, its more of an interface for the assessments-client or other components. I would imagine users would capture dates either through the UI or via the Excel template: in either case, I think the formatting or interpretation is handled for them. What do you think?

Reply:
As far as I know, the way Excel formats a date on the screen is dependent on your system date/format settings, but the underlying value is stored as a proper Date, if you know what I mean.